adagrams/game.py

```diff
@@ -1,13 +1,129 @@
 from random import randint
 
-def draw_letters():
-    pass
+def draw_letters(): # No parameters
+    
+    # This letter pool should reflect the distribution of letters as described in the Readme file Wave 1
+    distribution_of_letters = { "A": 9, "B": 2, "C": 2, "D": 4, "E": 12, "F": 2, "G": 3, "H": 2, "I": 9, "J": 1, "K": 1, "L": 4, "M": 2, "N": 6, "O": 8, "P": 2, "Q": 1, "R": 6, "S": 4, "T": 6, "U": 4, "V": 2, "W": 2, "X": 1, "Y": 2, "Z": 1 }
+    
+    letter_pool = [] # This represents the pool of letters that the player can draw from
 
-def uses_available_letters(word, letter_bank):
-    pass
+    for letter, frequency in distribution_of_letters.items():
+        for i in range(frequency):
+            letter_pool.append(letter)
+
+
+    # Each string should contain exactly one letter
+    # There are only 2 available C letters, so draw_letters cannot ever return more than 2 Cs
+    # Since there are 12 Es but only 1 Z, it should be 12 times as likely for the user to draw an E as a Z
+    # Invoking this function should not change the pool of letters
+    # Imagine that the user returns their hand to the pool before drawing new letters
+
+    hand = [] # These represent the hand of letters that the player has drawn
+    random_index = 0
+    for i in range(10):
```

Comment:
Notice that you declare the looping variable `i`, but don't use it in the body of your for loop. When we write a for loop, but don't need the looping variable, we should use `_` instead.

Also, if I did not know how to play Adagrams, I would not understand the significance of 10 here. If I were a dev who tried to read this, I might be wondering "Why do we only loop 10 times"?

In Python, `10` on this line is what is known as a "Magic Number". We use the term to refer to hard coded values that just kind of appear without any context to them!

It would be more descriptive to use a constant variable to reference 10 so that your code is more self-documenting.

```suggestion
    NUM_TILES_ALLOWED_IN_HAND = 10
    for _ in range(NUM_TILES_ALLOWED_IN_HAND):
```